### PR TITLE
Add Viewport Tests (and other test patches)

### PR DESCRIPTION
**Pull request by dgarciabriseno**
# Summary

Note: This will fail the first CI run because I'll have to update some snapshots

- Add test for centering the viewport on mobile.
- Update jsdoc comments
- Add mask to screenshot test to fix edge case where text says "2 seconds ago" instead of "1 second ago"
- Also update screenshot test to wait for screenshot to load before doing visual comparison (Previously this worked because we did "WaitForImgLoad" but we modified that function to only wait for img tiles, not all images).
- Update running/base difference test to wait for tiles to update before checking the difference value

### CODE REVIEW

File: tests/mobile/viewport.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  // 1. Center the viewport
+  await mobile.CenterViewport();
+  // 2. Expect the screenshot to match. Referencing by name so we can re-use it later.
+  const centered_aia_304_image = "centered_aia_304.png";
+  const opts: PageAssertionsToHaveScreenshotOptions = { mask: [page.locator('#time'), page.locator('#date')] };
+  await expect(page).toHaveScreenshot(centered_aia_304_image, opts);
+  // 3. Drag the sun off center
+  await mobile.moveViewport(250, 250);
+  // 4. expect the screenshot not to match
+  // await expect(page).not.toHaveScreenshot(centered_aia_304_image, opts);
```
Comment (dgarciabriseno) [inline tests/mobile/viewport.spec.ts]: Uncomment this
```suggestion
  await expect(page).not.toHaveScreenshot(centered_aia_304_image, opts);
```
